src/environment.rs

```diff
@@ -34,6 +43,87 @@ impl Environment {
         }
     }
 
+    /// Stores all driver description and attributes in a Vec
+    pub fn drivers(&self) -> Result<Vec<DriverInfo>> {
+        // Iterate twice, once for reading the maximum required buffer lengths so we can read
+        // everything without truncating and a second time for actually storing the values
+        let string_buf = std::ptr::null_mut();
+        let mut desc_length_out: raw::SQLSMALLINT = 0;
+        let mut attr_length_out: raw::SQLSMALLINT = 0;
+        let mut max_desc = 0;
+        let mut max_attr = 0;
+        let mut count = 0;
+        let mut result;
+        unsafe {
+            // Although the rather lengthy function call kind of blows the code, let's do the first
+            // one using SQL_FETCH_FIRST, so we list all drivers independent from environment state
+            result = raw::SQLDrivers(self.handle,
+                                     raw::SQL_FETCH_FIRST,
+                                     string_buf,
+                                     0,
+                                     &mut desc_length_out as *mut raw::SQLSMALLINT,
+                                     string_buf,
+                                     0,
+                                     &mut attr_length_out as *mut raw::SQLSMALLINT);
+        }
+        loop {
+            match result {
+                raw::SQL_SUCCESS |
+                raw::SQL_SUCCESS_WITH_INFO => {
+                    count += 1;
+                    max_desc = std::cmp::max(max_desc, desc_length_out);
+                    max_attr = std::cmp::max(max_attr, attr_length_out);
+                }
+                raw::SQL_NO_DATA => break,
+                raw::SQL_ERROR => return Err(Error {}),
+                _ => unreachable!(),
```

Comment:
That is intentional, let me motivate my choice a bit:
The only other thing that SQLDriver might return is SQL_INVALID_HANDLE (https://msdn.microsoft.com/en-us/library/ms712400(v=vs.85).aspx). We know it will not return this, if our invariant (our wrapped environment pointer is valid) holds true. So the path should actually be unreachable (provided the driver manager in C does not have a Bug and I did not make an error in my reasoning). If the invariant is violated, I think a panic is preferable to continued execution with an invalid `Environment`.